scripts/metadata_table.py

```diff
@@ -96,475 +125,1137 @@
+            if module:
+                perr = "module not allowed as argument when reading file"
+                raise ParseInternalError(perr)
+            # end if
+            if known_ddts is None:
+                known_ddts = list()
+            # end if
+            self.__start_context = ParseContext(context=self.__pobj)
+            self.__init_from_file(known_ddts, logger)
+        # end if
+
+    def __init_from_file(self, known_ddts, logger):
+        """ Read the table preamble, assume the caller already figured out
+        the first line of the header using the header_start method."""
+        curr_line, _ = self.__pobj.next_line()
+        in_properties_header = True
+        skip_rest_of_section = False
+        self.__dependencies = list() # Default is no dependencies
+        # Process lines until the end of the file or start of the next table.
+        while ((curr_line is not None) and
+               (not MetadataTable.table_start(curr_line))):
+            if MetadataSection.header_start(curr_line):
+                in_properties_header = False
+            # end if
+            if blank_metadata_line(curr_line):
+                curr_line, _ = self.__pobj.next_line()
+            elif in_properties_header:
+                # Process the properties in this table header line
+                for prop in _parse_config_line(curr_line, self.__pobj):
+                    # Manually parse name, type, and table properties
+                    key = prop[0].strip().lower()
+                    value = prop[1].strip()
+                    if key == 'name':
+                        self.__table_name = value
+                    elif key == 'type':
+                        if value not in TABLE_TYPES:
+                            self.__pobj.add_syntax_err("metadata table type",
+                                                       token=value)
+                        # end if
+                        self.__table_type = value
+                    elif key == 'dependencies':
+                        if value.lower() == "none" or value.strip() == '':
+                            if self.__dependencies:
+                                emsg = "dependencies = {} is ".format(value)
+                                depends = ", ".join(self.__dependencies)
+                                emsg += "incompatible with {}".format(depends)
+                                self.__pobj.add_syntax_err(emsg)
+                            else:
+                                self.__dependencies = None
+                            # end if
+                        elif self.__dependencies is None:
+                            emsg = "Cannot add dependencies, they have "
+                            emsg += "already been declared as None"
+                            self.__pobj.add_syntax_err(emsg)
+                        else:
+                            depends = [x.strip() for x in value.split(',') if x.strip()]
```

Comment:
This change should be made in feature/capgen **first**, then merged into prebuild. Doing it this way will cause issues later on.

Reply:
If there is a PR to feature/capgen that I can pull in that is identical to this change - and nothing else - then I can pull it in, fix the merge conflict, assert that nothing has changed in the codebase, and don't have to rerun the tests. I don't see however why this change is causing a problem. It can be merged back from main to feature/capgen without issues.

Reply:
See #395

Reply:
> I don't see however why this change is causing a problem. 

For one thing, it took me 20 times longer to process this PR due to this change and sloppy merging.